scrape.js

```diff
@@ -1,44 +1,53 @@
-const request = require("request")
+const request = require("request-promise")
 const cheerio = require("cheerio")
 const pretty = require("./pretty")
+const Promise = require("bluebird")
 
 const config = department =>
   ({
     method: "post",
-    url: "http://appl101.lsu.edu/booklet2.nsf/68a84f901daef98386257b43006b778a?CreateDocument",
+    uri: "http://appl101.lsu.edu/booklet2.nsf/68a84f901daef98386257b43006b778a?CreateDocument",
     body: `%25%25Surrogate_SemesterDesc=1&SemesterDesc=Spring+2017&%25%25Surrogate_Department=1&Department=${department}`
   })
 
-const departments = [ "COMPUTER SCIENCE" ]
+const scrape = (departments) => {
+
+  const departmentRequest = department =>
+    request(config(department))
+      .then(body => handleRequestResponse(body))
+
+  return Promise.map(
+    departments,
+    department => departmentRequest(department),
+    { concurrency: Infinity }
+  )
 
-const scrape = () => {
-  departments.forEach(department => request(config(department), onResponse))
 }
 
-const onResponse = (error, response, body) => {
-  if (error) {
-    console.log("Error: " + error)
-    return
-  }
+const handleRequestResponse = (body) => {
 
   let $ = cheerio.load(body)
-  let pre = $("pre")
+  let pre = $("pre").html()
+
+  if (!pre) {
+    //console.warn("Warning: Error on empty pre, most likely no courses in department")
```

Comment:
yeah probably not. Changed it to this

![image](https://cloud.githubusercontent.com/assets/1844458/22006128/455fdc18-dc2e-11e6-8ec4-97640f7d029a.png)